questionnaire/submissions.test.js

```diff
@@ -472,3 +489,46 @@ describe('Questionnaire submissions', () => {
         });
     });
 });
+
+describe('Questionnaire resubmissions', () => {
+    describe('Message Bus operational', () => {
+        it('should set a submission to IN_PROGRESS', async () => {
+            // eslint-disable-next-line global-require
+            const createQuestionnaireDAL = require('./questionnaire-dal');
+            const questionnaireDAL = createQuestionnaireDAL({logger: jest.fn()});
+            const createSubmissionsService = jest.requireActual('./submissions-service');
+            const submissionsService = createSubmissionsService({logger: jest.fn()});
+            await submissionsService.postFailedSubmissions();
+            expect(questionnaireDAL.updateQuestionnaireSubmissionStatus).toHaveBeenCalledWith(
+                '93259f6d-1826-4e97-ba39-53b4e232dd81',
+                'IN_PROGRESS'
+            );
+            expect(questionnaireDAL.updateQuestionnaireSubmissionStatus).not.toHaveBeenCalledWith(
+                '93259f6d-1826-4e97-ba39-53b4e232dd81',
+                'FAILED'
+            );
+        });
+    });
+    describe('Message Bus NOT operational', () => {
+        it('should set a submission to FAILED', async () => {
+            // eslint-disable-next-line global-require
+            const createQuestionnaireDAL = require('./questionnaire-dal');
+            const questionnaireDAL = createQuestionnaireDAL({logger: jest.fn()});
+            const createSubmissionsService = jest.requireActual('./submissions-service');
+            const submissionsService = createSubmissionsService({logger: jest.fn()});
+            await submissionsService.postFailedSubmissions();
+            expect(questionnaireDAL.updateQuestionnaireSubmissionStatus).toHaveBeenCalledWith(
+                '4ddb0208-f7da-4237-a244-34e7e58d2ddf',
+                'FAILED'
+            );
+        });
+    });
+    it('should resubmit failed submissions', async () => {
+        const response = await request(app)
+            .post('/api/v1/submissions/resubmit-failed')
+            .set('Authorization', `Bearer ${tokens['update:questionnaires']}`)
+            .set('Content-Type', 'application/vnd.api+json')
+            .send({});
+        expect(response.body).toEqual({status: 'ok'});
+    });
```

Comment:
This doesn't appear to assert "should resubmit failed submissions". Given you're returning from `postFailedSubmissions`, why not return that in the response and check that it's as expected?

Reply:
that is what I originally had and that was said to be not a good approach. So I removed a "useful" response.

Reply:
I can add it back in...